### ISSUE TITLE
Make selecting a conference when adding a submission more user friendly

### ISSUE THREAD

**Issue by nodunayo** [nodunayo/speakerline#15]
At the moment, this is the experience people have:

![screen shot 2017-03-05 at 18 50 13](https://cloud.githubusercontent.com/assets/4911302/23590348/f1abba1a-01d5-11e7-83ae-1c57b5d893da.png)

This makes it difficult to use and it's hard to work out whether the conference you are looking for has already been added or not.

Perhaps there's a way to type in a conference, select a year and then find or create the record depending on what's been entered?

Any other ideas welcome!

**Comment by deniseyu** [comment]
@nodunayo I've used this library before: https://harvesthq.github.io/chosen/ 

When you click in the dropdown menu, a textbox appears, and it will do an exact-string match against all the dropdown options. It's pretty great for filtering huge collections.

The drawback is it doesn't work on mobile browsers at all, the last time I tried.

**Comment by jemagee** [comment]
Hey - there’s a gem that does what I was going to try and write from scratch (in my mind i can see how it works with ajax and onKeyUp but just never done it, and testing JS is such a pain these days :)

**Comment by nodunayo** [comment]
Ooh, thanks for this @deniseyu!

Do you know what happens on mobile? Will it just be as if there was no library installed?

Thanks!

**Comment by deniseyu** [comment]
@nodunayo correct, I think the JS will become turned off on mobile and it will default to however the mobile browser handles dropdowns.

**Comment by andrew** [comment]
Slightly improved by https://github.com/nodunayo/speakerline/pull/23

![screen shot 2017-03-27 at 9 13 36 am](https://cloud.githubusercontent.com/assets/1060/24346765/bb798ca2-12cd-11e7-9f94-5b79bfcf71b3.png)

**Comment by jemagee** [comment]
Would you be opposed to including [redis](http://oldblog.antirez.com/post/autocomplete-with-redis.html) in the project?

**Comment by nodunayo** [comment]
Hmm, I'd have to do a bit more research on that. My instinct says avoid it unless you have to. 

Sorry for the slow response! I've been away.

**Comment by flyinggrizzly** [comment]
This is again slightly different after https://github.com/nodunayo/speakerline/pull/66

Event years are now grouped in optgroups

```
RailsConf
  2016
  2017
Bath Ruby
  2016
  2017
```

**Comment by rubydesign** [comment]
if you filtered out past ones and sort them in time order the list would be quite use-able until you have many more conferences in the db

**Comment by nodunayo** [comment]
@rubydesign — how would it work if you wanted to add a proposal to past events? In fact, the vast majority of the time, people are adding proposals in for past events, not upcoming ones.

**Comment by rubydesign** [comment]
@nodunayo Sorry, then i misunderstood the app. I thought people add proposals for talks they want to give in the future. Sorry

**Comment by nodunayo** [comment]
@rubydesign No worries at all. Thanks for trying to help! :-)

**Comment by rianrainey** [comment]
I'd like to use hotwire, if possible. I came across this [blog post](https://thoughtbot.com/blog/hotwire-essentials-is-here-learn-hotwire-by-building-a-podcast-player). Thoughts on doing something like [this](https://github.com/thoughtbot/botcasts/blob/main/lessons/lesson-4.md) where you typeahead and it filters the dropdown? 

Or are you okay doing a JS library? Might be easier. I see you still have jquery-rails so maybe you're not all in on hotwire just yet. This looks like a good list of vanilla js libraries to choose from: https://www.cssscript.com/best-autocomplete/ . Let us know which one you'd like to see implemented.
